src/test_fixture/ipa.rs

```diff
@@ -35,6 +35,60 @@ pub struct TestRawDataRecord {
     pub trigger_value: u32,
 }
 
+/// Executes IPA protocol in the clear, that is without any MPC helpers involved in the computation.
+/// Useful to validate that MPC output makes sense by comparing the breakdowns produced by MPC IPA
+/// with this function's results. Note that MPC version of IPA may apply DP noise to the aggregates,
+/// so strict equality may not work.
+///
+/// This function requires input to be sorted by the timestamp and returns a vector of contributions
+/// sorted by the breakdown key.
+///
+/// ## Panics
+/// Will panic if you run in on Intel 80286 or any other 16 bit hardware.
+pub fn ipa_in_the_clear(
+    input: &[TestRawDataRecord],
+    per_user_cap: u32,
+    attribution_window: u32,
+) -> Vec<u32> {
+    // build a view that is convenient for attribution. match key -> events sorted by timestamp in reverse
+    // that is more memory intensive, but should be faster to compute. We can always opt-out and
+    // execute IPA in place
```

Comment:
Rather than executing in place, we could iterate `input` once for each user.  That is, collect user IDs and then iterate `input` using `filter()`.  I seriously doubt that this will be a problem for test inputs.  This function takes many orders of magnitude less time than the MPC version right now.

Reply:
yea I was thinking about it too. What steered me away from this is that memory footprint is manageable: 10Mb overhead for 1M records and I did not need to change  `update_expected_output_for_user`.